doc/design/cluster_train/save_model.md

```diff
@@ -0,0 +1,100 @@
+# Design Doc: Save Model
+
+## Overview
+
+The model is the output of the training process. There are two
+ways from which user can obtain a model:
+
+- Save model triggered by user code: user code asks PaddlePaddle to
+  save a model.
+- Convert model from the snapshot: model being converted from
+  pservers' periodic snapshot. In this way, the user can cancel a job
+  at any time, and still have a relatively fresh model (we snapshot
+  around every 5 minutes).
+
+### Save Model Triggered by User Code
```

Comment:
I seems that this section descript the reason why we choice `trainer saving model`, so how about modify the title From 
`Save Model Triggered by User Code`
To
`Trainer Saving Model vs. Pservers Saving Model`